src/main/java/com/kodilla/ecommercee/entity/Cart.java

```diff
@@ -16,15 +16,15 @@
 public class Cart {
 
     @Id
-    @GeneratedValue
+    @GeneratedValue(strategy = GenerationType.AUTO)
     @EqualsAndHashCode.Include
     private Long id;
 
     @NonNull
-    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+    @OneToMany(cascade = CascadeType.MERGE)
     @JoinColumn(name = "cart_id")
```

Comment:
Od strony OneToMany JoinColumn jest niepotrzebne i robi dość nieprzewidziane rzeczy

Reply:
Po usunieciu joincolumn nie przechodziły mi testy. Zastosowałam sie do propozycji (sprawdziłam obydwie) i to też mi nie pomagało. Dlatego Joincolumn zostawiłam.

Reply:
Nie przechodzą, bo oprócz usunięcia @JoinColumn trzeba dodać mapowanie, brakuje mappedBy = "cart" -> '@OneToMany(mappedBy = "cart", cascade = CascadeType.REMOVE)' :) Jak to poprawisz, to całość śmiga. Niżej jeszcze jedna uwaga.

Reply:
Nie przechodzą, bo oprócz usunięcia @JoinColumn trzeba dodać mapowanie, brakuje mappedBy = "cart" -> '@OneToMany(mappedBy = "cart", cascade = CascadeType.REMOVE)' :) Jak to poprawisz, to całość śmiga. Niżej jeszcze jedna uwaga.